### DISCUSSION TITLE
Mlflow Evaluate: Custom Plot

### DISCUSSION THREAD

**Discussion by khojarohan4** [mlflow/mlflow#6377] [Q&A]
Trying to add a decile plot as a custom plot that can be called as a part of mlflow.evaluate. The decile plot needs to have the access to probabilities from the estimator so the eval_df isnt really helpful. From the examples I have seen on github for custom metrics and custom plots, I understand, I have to pass eval_df, builtin_metrics and the artificats_dir is optional. Anyway I can get access to the model's predict_proba method while I add this as a custom plot? or the right question could be, How do I achieve this?

**Comment by JHibbard** [comment]
Hi khojarohan4,

It is now possible to access `predict_proba` in [MLflow 1.29.0](https://github.com/mlflow/mlflow/pull/6631) using the `pyfunc_predict_fn` argument in log_model! Hopefully this access to `predict_proba` gets you over the hill to your solution:

```Python
import mlflow  # must be 1.29.0 or greater
from sklearn.datasets import load_wine
from xgboost import XGBClassifier


wine = load_wine()
X = wine.data
y = wine.target
 
with mlflow.start_run():
  xgb = XGBClassifier(random_state=0, use_label_encoder=False)
  xgb.fit(X, y)

  # Note: specify your prediction function here
  mlflow.sklearn.log_model(xgb, 'model', pyfunc_predict_fn='predict_proba', )


# Note: replace with your model location
logged_model = '...72220be6946/model'

# Load model as a PyFuncModel.
loaded_model = mlflow.pyfunc.load_model(logged_model)

# Predict on a Pandas DataFrame.
import pandas as pd
loaded_model.predict(pd.DataFrame(X))
```

Let me know if you need more help with your custom plotting as part of `mlflow.evaluate`, it'd be great to have you share your approach here for others too. This sounds like a common use case.

Thank you!
Jim
